hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -819,16 +819,21 @@ public MapConfig setOptimizeQueries(boolean optimizeQueries) {
     }
 
     private void validateSetOptimizeQueriesOption(boolean optimizeQueries) {
-        if (setCacheDeserializedValuesExplicitlyInvoked) {
-            if (optimizeQueries && cacheDeserializedValues == CacheDeserializedValues.NEVER) {
-                throw new ConfigurationException("Deprecated option 'optimize-queries' is set to true, "
-                        + "but 'cacheDeserializedValues' is set to NEVER. "
-                        + "These are conflicting options. Please remove the `optimize-queries'");
-            } else if (!optimizeQueries && cacheDeserializedValues == CacheDeserializedValues.ALWAYS) {
-                throw new ConfigurationException("Deprecated option 'optimize-queries' is set to false, "
-                        + "but 'cacheDeserializedValues' is set to ALWAYS. "
-                        + "These are conflicting options. Please remove the `optimize-queries'");
-            }
+        if (!setCacheDeserializedValuesExplicitlyInvoked) {
+            return;
+        }
+        if (cacheDeserializedValues == CacheDeserializedValues.INDEX_ONLY) {
```

Comment:
Is this really what we want? `INDEX_ONLY` is the default, so what makes calling `setCacheDeserializedValues(INDEX_ONLY)` so significant that it should suddenly trigger an error?